File: app/lib/models/matching/matching.dart
class Matching {
  final String requesterId;
  final String receiverId;
  final String status;
  final String preferredDate;
  final String location;
  final String message;
  final DateTime createdAt;
  final DateTime updatedAt;

  Matching({
    required this.requesterId,
    required this.receiverId,
    required this.status,
    required this.preferredDate,
    required this.location,
    required this.message,
    required this.createdAt,
    required this.updatedAt,
  });

  factory Matching.fromJson(Map<String, dynamic> json) => Matching(
        requesterId: json['requester_id'],
        receiverId: json['receiver_id'],
        status: json['status'],
        preferredDate: json['preferred_date'],
        location: json['location'],
        message: json['message'],
        createdAt: DateTime.parse(json['created_at']),
        updatedAt: DateTime.parse(json['updated_at']),
      );

  Map<String, dynamic> toJson() => {
        'requester_id': requesterId,
        'receiver_id': receiverId,
        'status': status,
        'preferred_date': preferredDate,
        'location': location,
        'message': message,
        'created_at': createdAt.toIso8601String(),
        'updated_at': updatedAt.toIso8601String(),
      };
}

File: functions/entity/matching.js
// matchings 컬렉션 (매칭 요청)
// 문서 ID: matchingId (자동 생성)
// {
//   requester_id: string, // traveler
//   receiver_id: string,  // local
//   status: 'pending' | 'accepted' | 'rejected',
//   preferred_date: string, // YYYY-MM-DD
//   location: string,
//   message: string,
//   created_at: timestamp,
//   updated_at: timestamp
// }

class Matching {
  constructor(data) {
    this.requester_id = data.requester_id;
    this.receiver_id = data.receiver_id;
    this.status = data.status; // 'pending' | 'accepted' | 'rejected'
    this.preferred_date = data.preferred_date;
    this.location = data.location;
    this.message = data.message;
    this.created_at = data.created_at;
    this.updated_at = data.updated_at;
  }
}

module.exports = Matching;
